config/initializers/rack-attack.rb

```diff
@@ -0,0 +1,123 @@
+class Rack::Attack
+  ### Configure Cache ###
+
+  # In this class you can set the following variables to configure the blocklisting, whitelisting,
+  # and throttling of the requests that the API receives:
+  # limit: amount of request that a user can make in a period without being blocked for the rest of
+  #        the period
+  # period: period of time for the previous limit. When this period is reached, you can make n
+  #         more requests, where n is the limit previously defined.
+  # maxretry: amount of requests that a user can make in findtime without being blocked for the bantime
+  # findtime: period of time for the previous maxretry.
+  # bantime: period of time that a user is blocked if exceeds the maxretry in a findtime.
+  # Be careful that the configuration for malicious users do not affect normal users  behaviour.
+
+  def self.production?
+    Rails.env.production?
+  end
+
+  # If you don't want to use Rails.cache (Rack::Attack's default), then
+  # configure it here.
+  #
+  # Note: The store is only used for throttling (not blacklisting and
+  # whitelisting). It must implement .increment and .write like
+  # ActiveSupport::Cache::Store
+
+  # Rack::Attack.cache.store = ActiveSupport::Cache::MemoryStore.new
+
+  ### Throttle Spammy Clients ###
+
+  # If any single client IP is making tons of requests, then they're
+  # probably malicious or a poorly-configured scraper. Either way, they
+  # don't deserve to hog all of the app server's CPU. Cut them off!
+  #
+  # Note: If you're serving assets through rack, those requests may be
+  # counted by rack-attack and this throttle may be activated too
+  # quickly. If so, enable the condition to exclude them from tracking.
+
+  # Throttle all requests by IP (60rpm)
+  #
+  # Key: "rack::attack:#{Time.now.to_i/:period}:req/ip:#{req.ip}"
+
+  # throttle('req/ip', limit: 50, period: 5.seconds) do |req|
+  #   req.ip # unless req.path.start_with?('/assets')
+  # end
+
+  ### Prevent Brute-Force Login Attacks ###
+
+  # The most common brute-force login attack is a brute-force password
+  # attack where an attacker simply tries a large number of emails and
+  # passwords to see if any credentials match.
+  #
+  # Another common method of attack is to use a swarm of computers with
+  # different IPs to try brute-forcing a password for a specific account.
+
+  # Throttle POST requests to /login by IP address
+  #
+  # Key: "rack::attack:#{Time.now.to_i/:period}:logins/ip:#{req.ip}"
+
+  throttle('logins/ip', limit: 5, period: 5.seconds) do |req|
```

Comment:
I would comment this and leave it as an example. Its not very generic. The same for the ones below.